microprofile/server/src/main/java/io/helidon/microprofile/server/JaxRsCdiExtension.java

```diff
@@ -0,0 +1,261 @@
+/*
+ * Copyright (c) 2019 Oracle and/or its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.microprofile.server;
+
+import java.util.HashSet;
+import java.util.LinkedHashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ExecutorService;
+import java.util.function.Supplier;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+import javax.enterprise.event.Observes;
+import javax.enterprise.inject.spi.Extension;
+import javax.enterprise.inject.spi.ProcessAnnotatedType;
+import javax.enterprise.inject.spi.WithAnnotations;
+import javax.ws.rs.Path;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.Application;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.ExceptionMapper;
+
+import io.helidon.common.HelidonFeatures;
+import io.helidon.common.HelidonFlavor;
+import io.helidon.webserver.jersey.JerseySupport;
+
+import org.eclipse.microprofile.config.Config;
+import org.eclipse.microprofile.config.ConfigProvider;
+import org.glassfish.jersey.server.ResourceConfig;
+
+/**
+ * Configure Jersey related things.
+ */
+public class JaxRsCdiExtension implements Extension {
+    private static final Logger LOGGER = Logger.getLogger(JaxRsCdiExtension.class.getName());
+
+    static {
+        HelidonFeatures.register(HelidonFlavor.MP, "JAX-RS");
+    }
+
+    private final List<JaxRsApplication> applicationMetas = new LinkedList<>();
+
+    private final Set<Class<? extends Application>> applications = new LinkedHashSet<>();
+    private final Set<Class<?>> resources = new HashSet<>();
+
+    private void collectApplications(@Observes ProcessAnnotatedType<? extends Application> applicationType) {
+        applications.add(applicationType.getAnnotatedType().getJavaClass());
+    }
+
+    private void collectResourceClasses(@Observes @WithAnnotations(Path.class) ProcessAnnotatedType<?> resourceType) {
+        Class<?> resourceClass = resourceType.getAnnotatedType().getJavaClass();
+        if (resourceClass.isInterface()) {
+            // we are only interested in classes - interface is most likely a REST client API
+            return;
```

Comment:
Not sure; doesn't Jersey let you define resource interfaces?  Maybe not.

Reply:
I have tried that even if I have an interface implemented by other resource classes, this would fail in Jersey.
This seems to be  reasonable way to filter them.